yt_dlp/extractor/netflix.py

```diff
@@ -0,0 +1,161 @@
+
+    }
+
+    def vprofile_to_vcodec(profile):
+        for key, value in vcodeclist.items():
+            if profile in value:
+                return key
+        return profile
+
+    def aprofile_to_acodec(profile, ext):
+        if "aac" in profile.lower():
+            return ("aac")
+        if "ddplus" in profile.lower():
+            return ("eac3")
+        if ext:
+            return "mp4"
+        else:
+            return(profile)
+
+    acodecs = ["heaac-2-dash", "heaac-5.1-dash", "heaac-2hq-dash", "xheaac-dash", "ddplus-2.0-dash", "ddplus-5.1-dash", "ddplus-atmos-dash"]
+    scodecs = ["simplesdh", "dfxp-ls-sdh", "webvtt-lssdh-ios8", "nflx-cmisc"]
+    url = []
+    a = 0
+    for vc in vcodecs:
+        vm = getVideoManifest(self, id, vcodecs[vc] + acodecs + scodecs, note="Trying Video Profiles: " + vc)
+        if "error" in vm:
+            continue
+        else:
+            if a == 0:
+                for langnode in reversed(traverse_obj(vm, ("result", "audio_tracks"), default={})):
+                    ac = try_get(langnode, lambda x: x["codecName"])
+                    for node in traverse_obj(langnode, ("streams"), default={}):
+                        aurl = {
+                            'url': url_or_none(try_get(node, lambda x: x["urls"][0]["url"])),
+                            'format_id': str_or_none(try_get(node, lambda x: x["downloadable_id"])),
+                            'abr': try_get(node, lambda x: x["bitrate"]),
+                            'ext': aprofile_to_acodec(ac or (try_get(node, lambda x: x["content_profile"])), ext=True),
+                            'filesize': try_get(node, lambda x: x["size"]),
+                            'acodec': aprofile_to_acodec(ac or (try_get(node, lambda x: x["content_profile"])), ext=False),
+                            'vcodec': "none",
+                            'language': try_get(node, lambda x: x["language"])
+                        }
+                        url.append(aurl)
+            a = 1
+        for node in traverse_obj(vm, ("result", "video_tracks", 0, "streams"), default={}):
+            vurl = {
+                'url': url_or_none(try_get(node, lambda x: x["urls"][0]["url"])),
+                'width': try_get(node, lambda x: x["res_w"]),
+                'height': try_get(node, lambda x: x["res_h"]),
+                'format_id': try_get(node, lambda x: x["downloadable_id"]),
+                'fps': try_get(node, lambda x: x["framerate_value"]),
+                'vbr': try_get(node, lambda x: x["bitrate"]),
+                'ext': "mp4",
+                'filesize': try_get(node, lambda x: x["size"]),
+                'vcodec': vprofile_to_vcodec(try_get(node, lambda x: x["content_profile"])),
+                'acodec': "none",
+                'quality': int_or_none(try_get(node, lambda x: x["vmaf"])),
+            }
+            url.append(vurl)
+    return(url)
```

Comment:
combine these into a NetflixBaseIE maybe

Reply:
Meh, I think this inheritance-based extractor design we, uh, inherited from upstream was something of a mistake in retrospect.

Reply:
It's still against yt-dlp's coding style convention, but there's no point in making a base class too, since it's just introducing only one class.

Reply:
functions/classes outside the extractor class are fine. But it doesn't make sense here. This class is even taking the IE as an argument. So it is best suited inside the IE